projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/components/DeployJob.tsx

```diff
@@ -48,46 +49,77 @@ export default class DeployJobDialog extends Component<IJobFullProps> {
       </>
     );
   }
+
+  handleCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const eventDetail = { runBeforeDeploy: event.target.checked };
+    const customEvent = new CustomEvent('deployRunChange', {
+      detail: eventDetail
+    });
+    window.dispatchEvent(customEvent);
+  };
 }
 
 export async function showCreateDeploymentDialog() {
+  let runBeforeDeploy = false;
+
+  const handleCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    runBeforeDeploy = event.target.checked;
+  };
+
   const result = await showDialog({
     title: CREATE_DEP_BUTTON_LABEL,
     body: (
-      <DeployJobDialog
-        jobName={jobData.get(VdkOption.NAME)!}
-        jobPath={jobData.get(VdkOption.PATH)!}
-        jobTeam={jobData.get(VdkOption.TEAM)!}
-      ></DeployJobDialog>
+      <>
+        <DeployJobDialog
+          jobName={jobData.get(VdkOption.NAME) || ''}
+          jobPath={jobData.get(VdkOption.PATH) || ''}
+          jobTeam={jobData.get(VdkOption.TEAM) || ''}
+        />
+        <div>
+          <input
+            type="checkbox"
+            name="deployRun"
+            id="deployRun"
+            className="jp-vdk-checkbox"
+            onChange={handleCheckboxChange}
+          />
```

Comment:
The checkbox should be checked by default. That is running the job prior to deployment is opt-out (not opt-in) . This is it encourage developers to always test and run their code prior to deployment.

Reply:
Because if you deploy using Notebook, you've developed it using notebooks. 

The running before deployment address some of the biggest issues with using notebooks - out of order cell executions, lack of reproducability/replicability . There's a nice podcast where a guy explain those (https://towardsdatascience.com/the-case-against-the-jupyter-notebook-d4da17e97243) 

So by running it before deployment with vdk run, we ensure the job would run the same way and finish as expected before deployment and after deployment in the cloud.

Reply:
changed it to checked by default